src/coreclr/jit/simdcodegenxarch.cpp

```diff
@@ -60,40 +60,70 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 // Return Value:
 //    None.
 //
-void CodeGen::genStoreIndTypeSIMD12(GenTree* treeNode)
+void CodeGen::genStoreIndTypeSimd12(GenTreeStoreInd* treeNode)
 {
-    assert(treeNode->OperGet() == GT_STOREIND);
+    assert(treeNode->OperIs(GT_STOREIND));
 
-    GenTree* addr = treeNode->AsOp()->gtOp1;
-    GenTree* data = treeNode->AsOp()->gtOp2;
+    // Should not require a write barrier
+    assert(gcInfo.gcIsWriteBarrierCandidate(treeNode) == GCInfo::WBF_NoBarrier);
 
-    // addr and data should not be contained.
-    assert(!data->isContained());
-    assert(!addr->isContained());
+    GenTree* addr = treeNode->Addr();
+    genConsumeAddress(addr);
 
-#ifdef DEBUG
-    // Should not require a write barrier
-    GCInfo::WriteBarrierForm writeBarrierForm = gcInfo.gcIsWriteBarrierCandidate(treeNode->AsStoreInd());
-    assert(writeBarrierForm == GCInfo::WBF_NoBarrier);
-#endif
+    GenTree*  data    = treeNode->Data();
+    regNumber dataReg = genConsumeReg(data);
 
-    // Need an additional Xmm register to extract upper 4 bytes from data.
-    regNumber tmpReg = treeNode->GetSingleTempReg();
+    emitter* emit = GetEmitter();
+
+    // Store lower 8 bytes
+    emit->emitInsStoreInd(INS_movsdsse2, EA_8BYTE, treeNode);
 
-    genConsumeOperands(treeNode->AsOp());
+    // Update the addr node to offset by 8
 
-    // 8-byte write
-    GetEmitter()->emitIns_AR_R(ins_Store(TYP_DOUBLE), EA_8BYTE, data->GetRegNum(), addr->GetRegNum(), 0);
+    if (treeNode->isIndirAddrMode())
+    {
+        GenTreeAddrMode* addrMode = addr->AsAddrMode();
+        addrMode->SetOffset(addrMode->Offset() + 8);
+    }
+    else if (addr->IsCnsIntOrI() && addr->isContained())
+    {
+        GenTreeIntConCommon* icon = addr->AsIntConCommon();
+        icon->SetIconValue(icon->IconValue() + 8);
```

Comment:
Are we guaranteed to never see relocatable handles here? It is not valid to `icon->IconValue() + 8` for them.

Reply:
Discussed this with Single over Discord. Pretty sure we can't hit this today, but might in the future with some planned NAOT optimizations.

It was trivial to workaround, however, and has been handled accordingly.

Reply:
I had thought relocatable handles were always blocked from containment, but that isn't true for indirs.